### PR TITLE
sc107109/action-comment-github-advisory-report

### PR DESCRIPTION
**Pull request by neo-anand-patel**
Description: Leaves a comment on the PR. This comment is a summary of vulnerabilities found in the packages and is returned from a endpoint

### CODE REVIEW

File: src/make-comment.ts [hunk 1]
```diff
@@ -35,19 +35,21 @@ const makeComment = async (commentData?: string): Promise<void> => {
     console.log('botComment', botComment);
     console.log('commentData', commentData);
 
+    const body = commentData ?? (await constructComment(commentData));
```
Comment (AD-Blue) [inline src/make-comment.ts:38]: would it be safer to pass nothing to `constructComment` instead of commentData when it can possibly be undefined?

**Review by AD-Blue — COMMENTED** [review]
Looks good, quick question/idea

File: src/index.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ const run = async (): Promise<void> => {
   try {
     const cloudFrontAuth = getInput('cloudFrontAuth');
     const serviceUrl = getInput('serviceUrl');
+    let generateReportResponse;
```
[the remark this answers is not in the document]
Comment (neo-anand-patel) [reply src/index.ts:15]: We assign the variable in a try catch when we call the `generateReport` method. But I need access to it outside of it as well when we call the `makeComment` method

File: src/make-comment.ts (hunk 1 above)
Comment (neo-anand-patel) [reply src/make-comment.ts:38]: yeah, good point

**Review by TimaJuma — APPROVED** [review]

**Review by AD-Blue — APPROVED** [review]

**Review by rogersnick — APPROVED** [review]

**Review by TimaJuma — APPROVED** [review]

**Review by AdamTranquilla — APPROVED** [review]

**Review by PeterHuang516 — APPROVED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -45,16 +46,16 @@ const run = async (): Promise<void> => {
...
-        console.log(report);
+        console.log(generateReportResponse);
       }
     } catch (error) {
       console.log(`${error}, Could not send data, printing comment`);
     }
 
     if (context.payload.pull_request) {
-      makeComment('');
+      makeComment(generateReportResponse?.report);
```
Comment (iansu) [inline src/index.ts:58]: If `generateReportResponse` is null then we probably don't want to make a comment at all

**Review by iansu — DISMISSED** [review]

**Review by AdamTranquilla — APPROVED** [review]

**Pull request merged**